catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswQueryFactory.java

```diff
@@ -232,6 +233,90 @@ private Filter transformCustomFunctionToFilter(Filter filter) {
         return filter;
     }
 
+    private QueryRequest normalizeSort(QueryRequest request) {
+        if (request == null) {
+            return null;
+        }
+
+        Query query = request.getQuery();
+        if (query == null) {
+            return request;
+        }
+
+        List<SortBy> sortBys = new ArrayList<>();
+
+        // Primary sort parameter
+        SortBy sortBy = query.getSortBy();
+        SortBy newSortBy = normalizeSortBy(sortBy);
+        if (newSortBy != null) {
+            sortBys.add(newSortBy);
+        }
+
+        // Additional sort parameters
+        Serializable extraSortBys = request.getPropertyValue(EXT_SORT_BY);
+        if (extraSortBys != null && extraSortBys instanceof SortBy[]) {
```

Comment:
Maybe it would it be better to make this instance check an assertion. Otherwise, if this property is ever changed to something besides a `SortBy[]`, it will _silently_ be unable to read the extra sort parameters.

Reply:
I believe `assert` shouldn't be used in production.  `null` is not an instance of `SortBy`, so I think you could remove the null check here.

Reply:
Fair enough about the assertion. Still, shouldn't there be some sort of error if `extraSortBys` is not `null` and not a `SortBy[]`?

Reply:
I think we tend to try to do the next best thing if any errors occur. In this case, the extra sort bys are silently ignored. I'll add a debug log message though for some kind of record.

Reply:
that sounds reasonable to me

Reply:
If it's at least logged, that should be fine.